relayer/connection-tx.go

```diff
@@ -218,6 +217,9 @@ func ExecuteConnectionStep(src, dst *Chain) (success, last, modified bool, err e
 			return false, false, false, err
 		}
 
+	case srcConn.Connection.State == conntypes.OPEN && dstConn.Connection.State == conntypes.OPEN:
+		last = true
+
```

Comment:
@colin-axner, Added this extra condition since both src connection and dst connection states are OPEN and going into infinite loop. Is this right way?

Reply:
yes it is! It fixes a bug I was running into previously as well. Great job, great intuition. I'm glad there are tests now. 

I redesigned the handshakes with the assumption that we were beginning with empty identifiers and therefore overlooked the case when a user supplies fully set identifiers